.buildkite/steps/conda-build.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+set -xeo pipefail
+exitCode=0
+
+echo "+++ :snake: :construction_worker: conda build"
+conda build . --no-anaconda-upload || exitCode=$?
+# if successful, then upload to buildkite
+if [ "${exitCode}" == "0" ]; then
+  conda_package="$(conda build . --output)"
+  buildkite-agent artifact upload "${conda_package}"
+fi
+
+exit $exitCode
```

Comment:
I think in this case, because there's no code to run if `conda build .` fails (unlike notebooks and tests where we want to upload the notebooks and JUnit xml files, respectively), we can just have this be straight line and rely on the `set -e` to have the script stop running if the build fails:

```suggestion
conda build . --no-anaconda-upload

echo "+++ :snake::buildkite: upload package"
conda_package="$(conda build . --output)"
buildkite-agent artifact upload "${conda_package}"
```